packages/modes/blockchain/mode.go

```diff
@@ -132,6 +130,16 @@ func (mode *Blockchain) Stop() {
 	log.Infoln("Blockchain mode stopped")
 }
 
+// API returns mode API
+func (mode *Blockchain) API() *httprouter.Router {
+	return mode.api
+}
+
+// Mode returns node type
+func (mode *Blockchain) Mode() modes.ModeType {
```

Comment:
I didn't like that Mode() method, because it exposes object outside, if you want to do some mode dependend operations, create one more method(such as smart.GetVM)